packages/transaction-controller/src/helpers/GasFeePoller.ts

```diff
@@ -54,13 +57,15 @@ export class GasFeePoller {
     gasFeeFlows,
     getEthQuery,
     getGasFeeControllerEstimates,
+    getProvider,
     getTransactions,
     layer1GasFeeFlows,
     onStateChange,
   }: {
     gasFeeFlows: GasFeeFlow[];
     getEthQuery: (chainId: Hex, networkClientId?: NetworkClientId) => EthQuery;
```

Comment:
Since an `EthQuery` instance can easily be created from a provider, ideally we only want one input here so we can remove `getEthQuery`.

Reply:
That's right, fixed in https://github.com/MetaMask/core/pull/4055/commits/f356ac5881371f3553795230f24d9e54a74dc701